src/components/molecules/Visualizer/Plugin/ModalContainer/index.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+  forwardRef,
+  ForwardRefRenderFunction,
+  useRef,
+  useImperativeHandle,
+  useCallback,
+} from "react";
+
+import theme, { styled } from "@reearth/theme";
+
+export type PluginModalInfo = {
+  id?: string;
+  background?: string;
+};
+
+type Props = {
+  showPluginModal: (modalInfo?: PluginModalInfo) => void;
+  shownPluginModalInfo?: PluginModalInfo;
+};
+
+const ModalContainer: ForwardRefRenderFunction<HTMLDivElement | undefined, Props> = (
+  { showPluginModal, shownPluginModalInfo },
+  ref,
+) => {
+  const innerRef = useRef<HTMLDivElement>(null);
+
+  useImperativeHandle(ref, () => innerRef.current as HTMLDivElement);
+
+  const closePopup = useCallback(() => {
```

Comment:
handlePopupClose

Reply:
Updated. Actually this should be `handleModalClose`. Sry for the typo.

Reply:
Haha glad we caught that :P